### ISSUE TITLE
Lint/UselessAccessModifier false positive for private_class_method

### ISSUE THREAD

**Issue by dduugg** [rubocop-hq/rubocop#6534]
## Expected behavior

Describe here how you expected RuboCop to behave in this particular situation.

I've noticed that rubocop erroneously flags `private_class_method` defs as `Lint/UselessAccessModifier` offenses when they are immediately defined after opening the class. For example, this code should not be flagged:
```ruby
class Foo
  private_class_method def self.bar
    'bar'
  end
end
```
## Actual behavior

Describe here what actually happened.

The example above is flagged as a `Lint/UselessAccessModifier` offense. Notably, this slight variation correctly does not trigger an offense:
```ruby
class Foo
  def self.foo
    'foo'
  end

  private_class_method def self.bar
    'bar'
  end
end
```

## Steps to reproduce the problem

This is extremely important! Providing us with a reliable way to reproduce
a problem will expedite its solution.

Run rubocop over the above example.

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler. Here's an example:

```
$ [bundle exec] rubocop -V
0.60.0 (using Parser 2.5.3.0, running on ruby 2.5.1 x86_64-darwin17)
```
